app/controllers/users_controller.rb

```diff
@@ -80,7 +85,14 @@ def destroy
   def promote
     @user = User.find(params[:id])
     role_id = Role.find_by(roletype: 'Admin').id
-    @user = User.update(role_id: role_id)
+    @user.update(role_id: role_id)
+    redirect_to users_panel_path
+  end
+
+  def demote
```

Comment:
great feature this prevents the case of an admin never being able to be removed. Like the simple method you used.